security/utils.py

```diff
@@ -32,17 +32,31 @@ def is_atty_string(s):
     return bool(re.search(r'^' + re.escape('\x1b[') + r'\d+m$', s))
 
 
+def truncate_newline_from_left(value, max_length):
+    if len(value) <= max_length:
+        return value
+
+    truncated_value = value[len(value) + 2 - (max_length):]
```

Comment:
+2 is added because of '…\n'

Reply:
There is also unnecessary `()` around `max_length`.